src/hoi4_world/countries/hoi4_country_converter.cpp

```diff
@@ -170,6 +178,144 @@ std::vector<hoi4::EquipmentVariant> DetermineActiveVariants(const std::vector<ho
 }
 
 
+std::optional<hoi4::Unit> MakeTemplate(const hoi4::DivisionTemplate division, std::vector<hoi4::Battalion>& battalions)
+{
+   hoi4::UnitCount required = division.GetBattalions();
+   for (const auto& [ut, str]: division.GetSupport())
+   {
+      required[ut] += str;
+   }
+   float total = 0.0F;
+   for (const auto& [ut, str]: required)
+   {
+      total += str;
+      float found = 0.0F;
+      for (const auto& battalion: battalions)
+      {
+         if (battalion.GetType() != ut)
+         {
+            continue;
+         }
+         found += battalion.GetStrength();
+      }
+      if (found < str)
+      {
+         return {};
+      }
+   }
+   if (total < tolerance)
+   {
+      // Avoid an infinite loop of making divisions that don't require any strength.
+      return {};
+   }
+   int equipment = 100;
+   int location = 11666;  // Vienna.
+   for (const auto& [ut, str]: required)
+   {
+      float needed = str;
+      for (auto& battalion: battalions)
+      {
+         if (battalion.GetType() != ut)
+         {
+            continue;
+         }
+         auto current = battalion.GetStrength();
+         if (current > needed)
+         {
+            current = needed;
+         }
+         battalion.AddStrength(-current);
+         needed -= current;
+         // Create division with equipment of worst battalion in it.
+         // Approximate but reasonable.
+         equipment = std::min(equipment, std::min(100, battalion.GetEquipmentScale()));
+         if (needed < tolerance)
+         {
+            location = battalion.GetLocation();
+            break;
+         }
+      }
+   }
+
+   std::erase_if(battalions, [](const hoi4::Battalion& cand) {
+      return cand.GetStrength() < tolerance;
+   });
+
+   return hoi4::Unit{division.GetName(), equipment, location};
+}
+
+std::vector<hoi4::Unit> ConvertArmies(const std::string& tag,
+    const mappers::UnitMapper& unit_mapper,
+    const vic3::Buildings& buildings,
+    const std::vector<hoi4::DivisionTemplate>& division_templates,
+    const hoi4::States& states)
+{
+   std::vector<hoi4::Battalion> battalions;
+   std::vector<hoi4::Unit> units;
+   for (const auto& [vic3_id, hoi4_id]: states.vic3_state_ids_to_hoi4_state_ids)
+   {
+      const auto itr = states.hoi4_state_ids_to_owner.find(hoi4_id);
+      if (itr == states.hoi4_state_ids_to_owner.end())
+      {
+         continue;
+      }
+      if (itr->second != tag)
+      {
+         continue;
+      }
+      const auto barracks = buildings.GetBuildingInState("building_barracks", vic3_id);
```

Comment:
nit:
use `vic3::BuildingType::Barracks` instead of raw string

Reply:
Done.